core/sdk-core/src/test/java/software/amazon/awssdk/core/sync/RequestBodyTest.java

```diff
@@ -104,4 +103,19 @@ public void emptyBytesConstructorHasCorrectContentType() {
         RequestBody requestBody = RequestBody.empty();
         assertThat(requestBody.contentType()).isEqualTo(Mimetype.MIMETYPE_OCTET_STREAM);
     }
+
+    @Test
+    public void remainingByteBufferConstructorOnlyRemainingBytesCopied() throws IOException {
+        ByteBuffer bb = ByteBuffer.allocate(4);
+        bb.putInt(42);
+        bb.flip();
+
+        bb.get();
+
+        RequestBody requestBody = RequestBody.fromRemainingByteBuffer(bb);
+        assertThat(requestBody.contentLength()).isEqualTo(bb.remaining());
+
+        byte[] requestBodyBytes = IoUtils.toByteArray(requestBody.contentStreamProvider().newStream());
+        assertThat(bb).isEqualTo(ByteBuffer.wrap(requestBodyBytes));
```

Comment:
Not keen on the way this test is written, it's vulnerable to bb being mutated, also the assertion reads kinda back to front usually we assert the thing that's under test rather than the thing being under test as the thing we compare to.

Reply:
Gotcha, I'll flip it.